File: Numerical_Methods/Solving Systems of Linear Equations/Gauss-Seidel Iterations/python_solutions/problem2.py
def gauss_seidel():
    # System of equations:
    # 3x - y + z = 4
    # -x + 4y + 2z = 5
    # x + 2y + 5z = 6
    
    x = y = z = 1.0  # Initial values
    tolerance = 0.001
    iteration = 0
    converged = False

    print("Gauss-Seidel Iteration Method - Problem 2")
    print(f"Initial values: x = {x}, y = {y}, z = {z}")
    print(f"Tolerance: {tolerance}")
    print("\nIterations:")

    while not converged:
        iteration += 1
        
        # Calculate new values using Gauss-Seidel formula
        x_new = (4 + y - z) / 3
        y_new = (5 + x_new - 2*z) / 4
        z_new = (6 - x_new - 2*y_new) / 5

        # Check for convergence
        if (abs(x_new - x) < tolerance and
            abs(y_new - y) < tolerance and
            abs(z_new - z) < tolerance):
            converged = True

        # Update values
        x, y, z = x_new, y_new, z_new

        # Print current iteration
        print(f"Iteration {iteration}: x = {x:.6f}, y = {y:.6f}, z = {z:.6f}")

    print(f"\nSolution converged after {iteration} iterations:")
    print(f"x = {x:.6f}")
    print(f"y = {y:.6f}")
    print(f"z = {z:.6f}")

if __name__ == "__main__":
    gauss_seidel() 

File: Numerical_Methods/Solving Systems of Linear Equations/Gauss-Seidel Iterations/java_solutions/Problem2.java
public class Problem2 {
    public static void main(String[] args) {
        // System of equations:
        // 3x - y + z = 4
        // -x + 4y + 2z = 5
        // x + 2y + 5z = 6
        
        double x = 1.0, y = 1.0, z = 1.0; // Initial values
        double xNew, yNew, zNew;
        double tolerance = 0.001;
        int iteration = 0;
        boolean converged = false;

        System.out.println("Gauss-Seidel Iteration Method - Problem 2");
        System.out.println("Initial values: x = " + x + ", y = " + y + ", z = " + z);
        System.out.println("Tolerance: " + tolerance);
        System.out.println("\nIterations:");

        while (!converged) {
            iteration++;
            
            // Calculate new values using Gauss-Seidel formula
            xNew = (4 + y - z) / 3;
            yNew = (5 + xNew - 2*z) / 4;
            zNew = (6 - xNew - 2*yNew) / 5;

            // Check for convergence
            if (Math.abs(xNew - x) < tolerance &&
                Math.abs(yNew - y) < tolerance &&
                Math.abs(zNew - z) < tolerance) {
                converged = true;
            }

            // Update values
            x = xNew;
            y = yNew;
            z = zNew;

            // Print current iteration
            System.out.printf("Iteration %d: x = %.6f, y = %.6f, z = %.6f%n", 
                            iteration, x, y, z);
        }

        System.out.println("\nSolution converged after " + iteration + " iterations:");
        System.out.printf("x = %.6f%n", x);
        System.out.printf("y = %.6f%n", y);
        System.out.printf("z = %.6f%n", z);
    }
} 